### PR TITLE
Feature add seats

### PR DESCRIPTION
**Pull request by bramvankooten**

### CODE REVIEW

File: src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+
+      .team-member {
+        cursor: pointer;
+      }
+    </style>
+
+    <lancie-ajax auto-fire id="ajaxGetAllSeats" refurl="seats" on-lancie-ajax="onGetAllSeats"></lancie-ajax>
+    <lancie-ajax auto-fire id="ajaxGetTeamTickets" refurl="tickets/teammembers" on-lancie-ajax="onGetTeamTickets"></lancie-ajax>
+    <lancie-ajax id="ajaxReserveSeat" method="POST" on-lancie-ajax="onReserveSeat"></lancie-ajax>
+    <lancie-ajax id="ajaxAddSeatRow" method="POST" body="[[getSeatGroupDTOBody(seatGroup,seatAmount)]]" on-lancie-ajax="onAddSeatRow" refurl = "seats"></lancie-ajax>
```
Comment (martijnjanssen) [inline src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html]:
```suggestion
    <lancie-ajax id="ajaxAddSeatRow" method="POST" body="[[getSeatGroupDTOBody(seatGroup,seatAmount)]]" on-lancie-ajax="onAddSeatRow" refurl="seats"></lancie-ajax>
```

File: src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+    <lancie-ajax id="ajaxRemoveSeats" method="DELETE" body="[[getSeatGroupDTOBody(removeGroup,removeAmount)]]" on-lancie-ajax="onRemoveSeats" refurl = "seats"></lancie-ajax>
```
Comment (martijnjanssen) [inline src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html]:
```suggestion
    <lancie-ajax id="ajaxRemoveSeats" method="DELETE" body="[[getSeatGroupDTOBody(removeGroup,removeAmount)]]" on-lancie-ajax="onRemoveSeats" refurl="seats"></lancie-ajax>
```

File: src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html [hunk 3]
```diff
@@ -0,0 +1,430 @@
...
+    <lancie-ajax id="ajaxLockSeat" method="POST" body="[[lockBoolean]]" on-lancie-ajax="onLockSeat"></lancie-ajax>
+    <lancie-ajax id="ajaxLockGroup" method="POST" body="[[lockBoolean]]" on-lancie-ajax="onLockGroup"></lancie-ajax>
+
+    <paper-input label="Search user" value="{{filterUser}}"></paper-input>
```
Comment (martijnjanssen) [inline src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html]: Not sure we need a search here, I'd vouch for removing it, this just duplicates features from the frontend which we don't need here.

File: src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html [hunk 4]
```diff
@@ -0,0 +1,430 @@
...
+              <div class="seats-group">
+                <template is="dom-repeat" items="[[_getSeatGroup(seats, item)]]">
+                  <lancie-admin-seat tickets="[[tickets]]" filter="[[filterUser]]" seat="[[item]]" user="[[user]]" fixed="[[fixed]]"></lancie-admin-seat>
+                </template>
+              </div>
+            </div>
+          </template>
+          <div class="seatmap-row">
+            <div class="add-seat" on-tap="openAddSeatRowDialog">+</div>
+            <div class="add-seat" style="margin-top: 10px;" on-tap="openSeatRemoveDialog">-</div>
```
Comment (martijnjanssen) [inline src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html]: Style in the css please :)

File: src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html [hunk 5]
```diff
@@ -0,0 +1,430 @@
...
+    <lancie-ajax auto-fire id="ajaxGetAllSeats" refurl="seats" on-lancie-ajax="onGetAllSeats"></lancie-ajax>
+    <lancie-ajax auto-fire id="ajaxGetTeamTickets" refurl="tickets/teammembers" on-lancie-ajax="onGetTeamTickets"></lancie-ajax>
+    <lancie-ajax id="ajaxReserveSeat" method="POST" on-lancie-ajax="onReserveSeat"></lancie-ajax>
+    <lancie-ajax id="ajaxAddSeatRow" method="POST" body="[[getSeatGroupDTOBody(seatGroup,seatAmount)]]" on-lancie-ajax="onAddSeatRow" refurl = "seats"></lancie-ajax>
+    <lancie-ajax id="ajaxRemoveSeats" method="DELETE" body="[[getSeatGroupDTOBody(removeGroup,removeAmount)]]" on-lancie-ajax="onRemoveSeats" refurl = "seats"></lancie-ajax>
+    <lancie-ajax id="ajaxLockSeat" method="POST" body="[[lockBoolean]]" on-lancie-ajax="onLockSeat"></lancie-ajax>
+    <lancie-ajax id="ajaxLockGroup" method="POST" body="[[lockBoolean]]" on-lancie-ajax="onLockGroup"></lancie-ajax>
+
+    <paper-input label="Search user" value="{{filterUser}}"></paper-input>
+    <div class="seatmapwrapper">
```
Comment (martijnjanssen) [inline src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html]: Change this name to `seatmap-wrapper`, the correct html casing with chaining words.

File: src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html [hunk 6]
```diff
@@ -0,0 +1,430 @@
...
+          <p>Seat is still available</p>
+        </div>
+      </div>
+      <div class="dialog-actions">
+        <paper-button on-tap="lockSeat">Toggle lock</paper-button>
+        <paper-button dialog-dismiss>Close</paper-button>
+      </div>
+    </lancie-dialog>
+
+    <lancie-dialog id=seatRowDialog>
```
Comment (martijnjanssen) [inline src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html]:
```suggestion
    <lancie-dialog id="seatRowDialog">
```

File: src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html [hunk 7]
```diff
@@ -0,0 +1,430 @@
...
+        <div class="dialog-content">
+          <lancie-error id="error2"></lancie-error>
+          <paper-input label="Group Name" value="{{seatGroup}}"></paper-input>
+          <paper-input label="Amount of seats" value="{{seatAmount}}"></paper-input>
+          <paper-button on-tap="addSeatRow">Add seats</paper-button>
+          <paper-button dialog-dismiss>Close</paper-button>
+        </div>
+    </lancie-dialog>
+
+    <lancie-dialog id=lockGroupDialog>
```
Comment (martijnjanssen) [inline src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html]:
```suggestion
    <lancie-dialog id="lockGroupDialog">
```

File: src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html [hunk 8]
```diff
@@ -0,0 +1,430 @@
...
+      <div class="dialog-actions">
+        <paper-button on-tap="lockSeat">Toggle lock</paper-button>
+        <paper-button dialog-dismiss>Close</paper-button>
+      </div>
+    </lancie-dialog>
+
+    <lancie-dialog id=seatRowDialog>
+        <h2>Add a row of seats</h2>
+        <div class="dialog-content">
+          <lancie-error id="error2"></lancie-error>
```
Comment (martijnjanssen) [inline src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html]: Let's not do this, give the element the correct name. `seatRowAddError` could be a suitable option here.

File: src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html [hunk 9]
```diff
@@ -0,0 +1,430 @@
...
+            <div class="add-seat" style="margin-top: 10px;" on-tap="openSeatRemoveDialog">-</div>
+          </div>
+        </div>
+      </div>
+    </div>
+
+    <lancie-dialog id="seatDialog">
+      <h2>Seat reservation</h2>
+      <div class="dialog-content">
+        <lancie-error id="error"></lancie-error>
```
Comment (martijnjanssen) [inline src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html]: `seatReservationError` could be a good name. Maybe a good idea to find a good format to handle all these duplicate elements with similar names, this only results in unclarity.

**Review by martijnjanssen — COMMENTED** [review]
I saw some issues, but I also skipped some. At some points you made mistakes regarding the naming of elements, but for the same elements chose a correct name. Could you look at the consistency in naming, consistency in position of cancel/confirm buttons, clearing forms/inputs? I also noticed that some `get...DTOBody`'s can be skipped by using a `lancie-form` instead of lancie-ajax, they work the same, might be nice to do in this new code.

But closing off: This feature is REALLY cool, this is one big step closer to a very complete adminpanel for running the events!

File: src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html (hunk 8 above)
Comment (bramvankooten) [reply src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html]: This error was not used at all, hence I removed it.

File: src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html (hunk 9 above)
Comment (bramvankooten) [reply src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-seats/lancie-admin-seats.html]: Resevation was not implemented on this page. I have removed reservation functions, and renamed the error to `editSeatError`

**Review by Sebastien1999 — APPROVED** [review]
The changes Martijn reuquested are for now all fixed. It looks good!

**Pull request merged**